comictaggerlib/ctsettings/__init__.py

```diff
@@ -25,3 +34,115 @@
     "ct_ns",
     "group_for_plugin",
 ]
+
+
+class SettingsEncoder(json.JSONEncoder):
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, pathlib.Path):
+            return {"type": "pathlib.Path", "value": str(obj)}
+
+        # Let the base class default method raise the TypeError
+        return json.JSONEncoder.default(self, obj)
+
+
+class SettingsDecoder(json.JSONDecoder):
+    def __init__(self, definitions: settngs.Definitions, **kwargs):
+        kwargs["object_hook"] = self.object_hook
+        super().__init__(**kwargs)
+        self.definitions = definitions
+
+    def object_hook(self, obj):
+        for o, v in obj.items():
+            # The source name is required, use the object with the full file
+            if o.startswith("Source ") and isinstance(v, dict):
+                for key, value in v.items():
+                    if isinstance(value, dict) and value.get("type"):
+                        typ = self.definitions[o].v[key].type
+                        if typ:
+                            # Should check types match?
+                            obj[o][key] = typ(value["value"])
+
+        return obj
+
+
+def parse_file(definitions: settngs.Definitions, filename: pathlib.Path) -> tuple[settngs.Config[settngs.Values], bool]:
+    """
+    Read options from a json dictionary from a file.
+
+    Args:
+        definitions: A set of setting definitions. See `Config.definitions` and `Manager.definitions`
+        filename: A pathlib.Path object to read a json dictionary from
+    """
+    options: settngs.Values = {}
+    success = True
+    if filename.exists():
+        try:
+            with filename.open() as file:
+                opts = json.load(file, cls=SettingsDecoder, definitions=definitions)
+            if isinstance(opts, dict):
+                options = opts
+            else:  # pragma: no cover
+                raise Exception("Loaded file is not a JSON Dictionary")
+        except Exception:  # pragma: no cover
+            logger.exception("Failed to load config file: %s", filename)
+            success = False
+    else:
+        logger.info("No config file found")
+        success = True
+
+    return settngs.normalize_config(settngs.Config(options, definitions), file=True), success
+
+
+def parse_config(
+    definitions: settngs.Definitions,
+    description: str,
+    epilog: str,
+    config_path: pathlib.Path,
+    args: list[str] | None = None,
+) -> tuple[settngs.Config[settngs.Values], bool]:
```

Comment:
This should take in the entire manager, there is no reason to list every individual argument here